pkg/provider/aws/iam.go

```diff
@@ -179,6 +179,7 @@ func RefreshIAMPolicy(awsClient Client, federatedRole *awsv1alpha1.AWSFederatedR
 
 	// Detach all removed polices from role
 	for _, policyArn := range policyArnsToDetach {
+		policyArn := policyArn // fix for implicit memory aliasing
```

Comment:
Yes, I tried the fix beneath and this led to a failing unit test. The error was an out-of-bounds error, so we tried to access an element at an index, that did not exist. I did not investigate this further and just opted for this solution here, because I do not wanted to change the tests.

Changing the tests, might change the desired behavior

Reply:
What about changing the later for loop to use this pattern?

Reply:
This should work